internal/network/command_create.go

```diff
@@ -0,0 +1,170 @@
+	pcmd.AddOutputFlag(cmd)
+
+	cobra.CheckErr(cmd.MarkFlagRequired("cloud"))
+	cobra.CheckErr(cmd.MarkFlagRequired("region"))
+	cobra.CheckErr(cmd.MarkFlagRequired("connection-types"))
+
+	return cmd
+}
+
+func (c *command) create(cmd *cobra.Command, args []string) error {
+	name := args[0]
+
+	cloud, err := cmd.Flags().GetString("cloud")
+	if err != nil {
+		return err
+	}
+	cloud = strings.ToUpper(cloud)
+
+	region, err := cmd.Flags().GetString("region")
+	if err != nil {
+		return err
+	}
+
+	connectionTypes, err := cmd.Flags().GetStringSlice("connection-types")
+	if err != nil {
+		return err
+	}
+
+	connectionTypesItems := make([]string, len(connectionTypes))
+	for i, ct := range connectionTypes {
+		connectionTypesItems[i] = strings.ToUpper(ct)
+	}
+
+	cidr, err := cmd.Flags().GetString("cidr")
+	if err != nil {
+		return err
+	}
+
+	zones, err := cmd.Flags().GetStringSlice("zones")
+	if err != nil {
+		return err
+	}
+
+	zoneInfo, err := cmd.Flags().GetStringSlice("zone-info")
+	if err != nil {
+		return err
+	}
+	zoneInfoItems, err := getZoneInfoItems(zoneInfo)
+	if err != nil {
+		return err
+	}
+
+	dnsResolution, err := cmd.Flags().GetString("dns-resolution")
+	if err != nil {
+		return err
+	}
+	dnsResolution = strings.ToUpper(dnsResolution)
+
+	reservedCidr, err := cmd.Flags().GetString("reserved-cidr")
+	if err != nil {
+		return err
+	}
+
+	environmentId, err := c.Context.EnvironmentId()
+	if err != nil {
+		return err
+	}
+
+	createNetwork := networkingv1.NetworkingV1Network{
+		Spec: &networkingv1.NetworkingV1NetworkSpec{
+			DisplayName:     networkingv1.PtrString(name),
+			Cloud:           networkingv1.PtrString(cloud),
+			Region:          networkingv1.PtrString(region),
+			ConnectionTypes: &networkingv1.NetworkingV1ConnectionTypes{Items: connectionTypesItems},
+			Environment:     &networkingv1.ObjectReference{Id: environmentId},
+		},
+	}
+
+	if cidr != "" {
+		createNetwork.Spec.SetCidr(cidr)
+	}
+
+	if len(zones) != 0 {
+		createNetwork.Spec.SetZones(zones)
+	}
+
+	if len(zoneInfoItems) != 0 {
+		createNetwork.Spec.SetZonesInfo(networkingv1.NetworkingV1ZonesInfo{Items: zoneInfoItems})
+	}
+
+	if dnsResolution != "" {
+		createNetwork.Spec.SetDnsConfig(networkingv1.NetworkingV1DnsConfig{Resolution: dnsResolution})
+	}
+
+	if reservedCidr != "" {
+		createNetwork.Spec.SetReservedCidr(reservedCidr)
+	}
+
+	network, err := c.V2Client.CreateNetwork(createNetwork)
+	if err != nil {
+		return err
+	}
+
+	return printTable(cmd, network)
+}
+
+func getZoneInfoItems(zoneInfo []string) ([]networkingv1.NetworkingV1ZoneInfo, error) {
+	zoneInfoItems := make([]networkingv1.NetworkingV1ZoneInfo, len(zoneInfo))
+	for i, info := range zoneInfo {
+		zoneInfo := strings.Split(info, "=")
+		if len(zoneInfo) != 2 {
+			return nil, errors.NewErrorWithSuggestions("invalid zones-info", `Specify a comma-separated list of "zone=cidr" pairs. For exmaple: "--zone-info usw2-az1=10.10.0.0/27,usw2-az3=10.10.0.32/27,usw2-az4=10.10.0.64/27"`)
```

Comment:
```suggestion
			return nil, errors.NewErrorWithSuggestions("invalid zones-info", `Specify a comma-separated list of "zone=cidr" pairs. For example: "--zone-info usw2-az1=10.10.0.0/27,usw2-az3=10.10.0.32/27,usw2-az4=10.10.0.64/27"`)
```

Reply:
Usually, we add examples in the `Example` field of the command.

Reply:
Given that we have already had an example demonstrating the usage of `--zone-info`. Would you say returning an error with the error message "invalid zone-info" is sufficient?

Reply:
Added a new comment to improve the error message!